templates/contract-deploy/run-contract-setup.sh

```diff
@@ -107,26 +107,38 @@ fund_account_on_l1 "l1testing" "{{.zkevm_l2_l1testing_address}}"
 
 echo_ts "Setting up local zkevm-contracts repo for deployment"
 pushd /opt/zkevm-contracts || exit 1
-# setup a foundry toml in case we do a gas token or dac deployment
-printf "[profile.default]\nsrc = 'contracts'\nout = 'out'\nlibs = ['node_modules']\n" > foundry.toml
-
 cp /opt/contract-deploy/deploy_parameters.json /opt/zkevm-contracts/deployment/v2/deploy_parameters.json
 cp /opt/contract-deploy/create_rollup_parameters.json /opt/zkevm-contracts/deployment/v2/create_rollup_parameters.json
 sed -i 's#http://127.0.0.1:8545#{{.l1_rpc_url}}#' hardhat.config.ts
 
 # Deploy gas token
 # TODO in the future this should be configurable. I.e. we should be able to specify a token address that has already been deployed
-# {{if .zkevm_use_gas_token_contract}}
+# {{if .gas_token_enabled}}
+
+# {{if eq .gas_token_address ""}}
 echo_ts "Deploying gas token to L1"
+printf "[profile.default]\nsrc = 'contracts'\nout = 'out'\nlibs = ['node_modules']\n" > foundry.toml
```

Comment:
This is a random perspective, but I would prefer to leave this where it was. I think in some of our docs, we assume that the `foundry.toml` has been created..

Reply:
Addressed in [dfcf931](https://github.com/0xPolygon/kurtosis-cdk/pull/386/commits/dfcf9315e3d54fb4627c6e42ca4b64f9c81469ec)